client/components/marketing-survey/cancel-jetpack-form/index.tsx

```diff
@@ -225,15 +225,27 @@ const CancelJetpackForm: React.FC< Props > = ( {
 		}
 	};
 
-	const setSurveyStep = ( stepFunction: ( step: string, steps: string[] ) => string ) => {
-		const newStep = stepFunction( cancellationStep, availableSurveySteps );
+	const setSurveyStep = useCallback(
+		( stepFunction: ( step: string, steps: string[] ) => string ) => {
+			const newStep = stepFunction( cancellationStep, availableSurveySteps );
 
-		setCancellationStep( newStep );
+			setCancellationStep( newStep );
 
-		// record tracks event
-		// since the steps used for the Jetpack cancellation flow are different, use a different event name for Tracks
-		recordEvent( 'calypso_purchases_cancel_jetpack_survey_step', { new_step: newStep } );
-	};
+			// record tracks event
+			// since the steps used for the Jetpack cancellation flow are different, use a different event name for Tracks
+			recordEvent( 'calypso_purchases_cancel_jetpack_survey_step', { new_step: newStep } );
+		},
+		[ availableSurveySteps, cancellationStep, recordEvent ]
+	);
+
+	// Disable continuation button if "Other" is selected but no reason is written in
+	useEffect( () => {
+		if ( surveyAnswerId === 'another-reason' && surveyAnswerText === '' ) {
```

Comment:
What do you think about doing something like
`if ( surveyAnswerId === 'another-reason' && surveyAnswerText.trim() === '' )`
in order to avoid people leaving blank spaces as well?

Reply:
Good thinking, I'll add that